code/game/gamemodes/objective.dm

```diff
@@ -164,6 +164,35 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return FALSE
 	return TRUE
 
+/datum/objective/assassinateonce
+	name = "Assassinate once"
+	martyr_compatible = TRUE
+	var/won = FALSE
+
+/datum/objective/assassinateonce/find_target(list/target_blacklist)
+	..()
+	if(target?.current)
+		explanation_text = "Teach [target.current.real_name], the [target.assigned_role], a lesson they will not forget. The target only needs to die once for success."
+		RegisterSignal(target, COMSIG_MOB_DEATH, PROC_REF(check_midround_completion))
```

Comment:
You also need another signal for if the target changes bodies, should look something like this
```
		RegisterSignal(target, COMSIG_MIND_TRANSER_TO, PROC_REF(target_changed_bodies))
```
However, this signal transfer to only sends the new body, so you'll need to alter it
![image](https://github.com/ParadiseSS13/Paradise/assets/91113370/7678ff1d-d3e4-4dd6-ab4d-53591b20880e) (mind.dm)
into something like
![image](https://github.com/ParadiseSS13/Paradise/assets/91113370/7e964f42-db1e-4063-af57-c4742f9eaa4c) (mind.dm, altered)
then with your new proc, do something like this
```
/datum/objective/assassinateonce/proc/target_changed_bodies(mob/living/new_character, mob/living/old_character)
	UnregisterSignal(old_character, COMSIG_MOB_DEATH)
	RegisterSignal(new_character, COMSIG_MOB_DEATH, PROC_REF(check_midround_completion))
```

Reply:
wait if they which bodies wouldn't the objective text need to update?